segments/dir.p9k

```diff
@@ -96,10 +115,14 @@ prompt_dir() {
         # for each parent path component find the shortest unique beginning
         # characters sequence. Source: https://stackoverflow.com/a/45336078
         if (( ${#current_path} > 1 )); then # root and home are exceptions and won't have paths
-          # cheating here to retain ~ as home folder
-          local home_path="$(getUniqueFolder $HOME)"
-          trunc_path="$(getUniqueFolder $PWD)"
-          [[ $current_path == "~"* ]] && current_path="~${trunc_path//${home_path}/}" || current_path="/${trunc_path}"
+          local cwd="current_path"
+          # 12 for "current_path", 2 characters in ":h", $LEN for setable variable
+          local cwd="${(r#$((12+2*${POWERLEVEL9K_SHORTEN_DIR_LENGTH}))##:h#)cwd}"
+          # split paths and shorten left one (beware the /)
+          local left_path="${${(P)cwd}}"
```

Comment:
I think [changes to `(P)`](https://github.com/zsh-users/zsh/blob/master/ChangeLog#L6715) since zsh 5.2 make this possible. I could implement a workaround with `if [[ "$(zsh --version)" == *"5.1.1"* ]] ...`, a for-loop and `dirame`...
This is what I started with. It would replace these lines in function:
```
local cwd="current_path"
local cwd="${(r#$((12+2*${POWERLEVEL9K_SHORTEN_DIR_LENGTH}))##:h#)cwd}"
local left_path="${${(P)cwd}}"
```